Common/Cutscenes/DialogueState.cs

```diff
@@ -9,101 +9,189 @@
 using Terraria.GameContent.UI.Elements;
 using ReLogic.Content;
 using OvermorrowMod.Core;
+using System.Text;
+using Terraria.GameContent;
+using System;
+using ReLogic.Graphics;
+using Terraria.UI.Chat;
 
 namespace OvermorrowMod.Common.Cutscenes
 {
     public class DialogueState : UIState
     {
-        private UIElement DialogueBox;
-        private UIText Name;
-        private UIText Dialogue;
-        public UIImage BackDrop;
-
         private int DialogueTimer;
         private int SecondaryTimer;
-
-        public override void OnInitialize()
-        {
-            DialogueBox = new UIElement();
-            DialogueBox.Width.Set(600f, 0f);
-            DialogueBox.HAlign = .5f;
-            DialogueBox.Top.Set(169, 0f);
-
-            BackDrop = new UIImage(ModContent.Request<Texture2D>(AssetDirectory.Textures + "GamerTag"));
-            BackDrop.Left.Set(0, 0f);
-            BackDrop.Top.Set(0, 0f);
-
-            Name = new UIText("", 1f);
-            Name.Top.Set(20, 0f);
-            Name.Left.Set(130, 0f);
-
-            Dialogue = new UIText("", 1f);
-            Dialogue.Top.Set(60, 0f);
-            Dialogue.Left.Set(143, 0f);
-
-            DialogueBox.Append(BackDrop);
-            DialogueBox.Append(Name);
-            DialogueBox.Append(Dialogue);
-            Append(DialogueBox);
-        }
+        private int OpenTimer;
+        private int CloseTimer;
+        private int DelayTimer;
 
         // This determines whether the UI is shown or not
         public override void Draw(SpriteBatch spriteBatch)
         {
-            if (Main.LocalPlayer.GetModPlayer<DialoguePlayer>().DialogueList.Count > 0)
-            {
-                BackDrop.Draw(spriteBatch);
-                Name.Draw(spriteBatch);
-                Dialogue.Draw(spriteBatch);
-            }
+            float OPEN_TIME = 15;
+            float CLOSE_TIME = 10;
+            float MAXIMUM_LENGTH = 280;
+            float DIALOGUE_DELAY = 30;
 
-            /*int xPosition = (int)(Main.screenWidth * Main.UIScale) / 2;
-            int yPosition = Main.screenHeight - 100;
+            int xPosition = 200;
+            int yPosition = Main.screenHeight - 375/*169*/;
 
-            Texture2D backDrop = ModContent.Request<Texture2D>(AssetDirectory.Textures + "GamerTag").Value;
-            spriteBatch.Draw(backDrop, new Vector2(xPosition + 65, yPosition), null, Color.White, 0f, backDrop.Size() / 2, 1f, SpriteEffects.None, 1f);
+            Vector2 textPosition = new Vector2(xPosition - 95, yPosition - 25);
 
-            base.Draw(spriteBatch);*/
-        }
-
-        // This handles the dialogue that the player has, if it detects that the player has new dialogue then it starts drawing it
-        public override void Update(GameTime gameTime)
-        {
             DialoguePlayer player = Main.LocalPlayer.GetModPlayer<DialoguePlayer>();
 
             if (player.DialogueList.Count > 0)
             {
```

Comment:
This method is a bit of a pain. In general, if you ever find yourself with

```
if (...) {
    // 30+ lines
}
```

Consider either inverting the condition:

`if (!...) return;`

or putting the contents in a different method.